### PR TITLE
Fix reinstallation of openmpi/pydusa

### PR DESCRIPTION
**Pull request by mstabrin**
Reinstalling OpenMPI on CentOS 7 caused a conda update which broke dependencies.

This way it works and in addition alows the usage of the systems OpenMPI

### CODE REVIEW

File: utils/build_and_install_openmpi.sh [hunk 1]
```diff
@@ -5,5 +5,17 @@
 set -xe
 
 RECIPES_DIR=$(cd $(dirname $0)/../recipes && pwd -P)
-conda build ${RECIPES_DIR}/openmpi -c defaults -c conda-forge
-conda install openmpi --use-local --yes
+conda build ${RECIPES_DIR}/openmpi -c defaults -c conda-forge -c conda-forge/label/cf201901
```
Comment (shadowwalkersb) [inline utils/build_and_install_openmpi.sh]: Are there any error messages that motivated adding ` -c conda-forge/label/cf201901`?

Comment (mstabrin) [reply utils/build_and_install_openmpi.sh]: Yes, the openmpi installation tries to install toolchain, which is not available in the other channels.

File: utils/build_and_install_openmpi.sh [hunk 2]
```diff
@@ -5,5 +5,17 @@
...
+
+CONDA_BLD=$(dirname $(which conda))/../conda-bld
+if [[ ! -d ${CONDA_BLD} ]]
```
[the remark this answers is not in the document]
Comment (mstabrin) [reply utils/build_and_install_openmpi.sh]: Conda seems to have a problem finding packages locally build with conda-build.
Therefore, I needed to set the channel manually to the anaconda/conda-bld directory.
However, this directory does not belong to the virtual environment, but the root directory.
This check checks, if the current directory is a virtual environment or the base environment of the anaconda installation.
If it cannot be found, the build will fail.

File: utils/install_pydusa_numpy.sh [hunk 3]
```diff
@@ -14,8 +14,19 @@ set -xe
...
+    CONDA_BLD=$(dirname ${CONDA_EXE})/../conda-bld
+fi
+if [[ ! -d ${CONDA_BLD} ]]
+then
+    echo "Conda-bld not found"
+    exit 1
+fi
+
+echo $(conda remove fftw-mpi --force --yes)
+conda install fftw-mpi --yes --override-channels -c file://${CONDA_BLD}
```
[the remark this answers is not in the document]
Comment (mstabrin) [reply utils/install_pydusa_numpy.sh]: Let me check this.

**Comment by mstabrin** [comment]
@shadowwalkersb Removing the removal seems to work just fine :)

File: utils/build_and_install_openmpi.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (shadowwalkersb) [reply utils/build_and_install_openmpi.sh]: I don't think testing for `conda-bld`'s existence is necessary here. If it doesn't exist the next conda-install command will just fail anyways.

To find `conda-bld`, I would use `conda info --root`. `which conda` may not be reliable. In some versions `conda` is not in `PATH` and is a shell function. Then, `which conda` doesn't return anything. So, this should be safer:
`CONDA_BLD="$(conda info --root)/conda-bld"`

File: utils/install_pydusa_numpy.sh [hunk 4]
```diff
@@ -14,8 +14,19 @@ set -xe
...
+then
+    CONDA_BLD=$(dirname ${CONDA_EXE})/../conda-bld
+fi
+if [[ ! -d ${CONDA_BLD} ]]
+then
+    echo "Conda-bld not found"
+    exit 1
+fi
+
+#echo $(conda remove fftw-mpi --force --yes)
```
Comment (shadowwalkersb) [inline utils/install_pydusa_numpy.sh]: Can you remove the whole line instead of commenting it out?

Comment (mstabrin) [reply utils/install_pydusa_numpy.sh]: Agreed

**Pull request merged**